src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java

```diff
@@ -377,12 +381,32 @@ protected void addAdditionalContextValues(
         String extensionUniqueId = getThreadContext().getHeader("extension_unique_id");
         if (extensionUniqueId != null) {
             ExtensionsManager extManager = OpenSearchSecurityPlugin.GuiceHolder.getExtensionsManager();
-            if (extManager.lookupExtensionSettingsById(extensionUniqueId).isPresent()) {
+            Optional<ExtensionsSettings.Extension> extension = extManager.lookupExtensionSettingsById(extensionUniqueId);
+
+            if (extension.isPresent() && isExtensionAllowed(extension.get().getDistinguishedNames(), principal)) {
```

Comment:
Hey @MaciejMierzwa, its possible to define the distinguished names setting from the security plugin through the [`ExtensionAwarePlugin` interface](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/plugins/ExtensionAwarePlugin.java). 

See an example here: https://github.com/cwperks/security/pull/5/files#diff-e7ef66295cba81a49e4349781a2e9678d2b021a570e978bb4feb422b03d5d74aR1133-R1139

IMO I think the definition of a setting should be in the same codebase as the implementation for it. Can this PR make the `OpenSearchSecurityPlugin` also implement `ExtensionAwarePlugin` and define a list setting similar to the link above?

Any settings added via `getExtensionSettings()` would be available by calling `extension.get().getAdditionSettings.get(<Setting>)`. See example [here](https://github.com/cwperks/security/pull/5/files#diff-5fce73d989856a9476f3c40b358f26887f92b05d87696d2c2edd5a6b8e1e4a37R170-R172)